module_dependencies/module/session.py

```diff
@@ -266,4 +272,8 @@ def post(
         payload = self.construct_payload(
             module, count=count, timeout=timeout, language=language
         )
-        return super().post(self.url, json=payload)
+        if self.token is not None:
+            logger.info('Using Sourcegraph API token')
+            return super().post(self.url, json=payload, headers={"Authorization": f"token {self.token}"})
+        else:
+            return super().post(self.url, json=payload)
```

Comment:
```suggestion
        return super().post(self.url, json=payload)
```
We can simplify this slightly.